projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.html

```diff
@@ -4,11 +4,14 @@
     [id]="_toggleId"
     class="lg-accordion__heading-toggle"
     [class.lg-accordion__heading-toggle--active]="_isActive"
+    [class.lg-accordion__heading-with-decorative-icon]="_hasDecorativeIcon"
     [attr.aria-expanded]="_isActive"
     [attr.aria-controls]="_panelId"
     (click)="toggle()"
   >
+    <ng-content select="lg-icon"></ng-content>
     <ng-content></ng-content>
-    <lg-icon class="lg-accordion__heading-icon" name="chevron-down"></lg-icon>
+    <lg-icon class="lg-accordion__heading-icon lg-accordion__heading-icon--animated" [ngClass]="{'lg-accordion__heading-icon--right': _hasDecorativeIcon}" name="chevron-down">
```

Comment:
If you wrap the chevron icon in a div, you can target/select/distinguish the added icon by ` > lg-icon` and you don't need to rely on `ngAfterInit` to find out if there's an icon or not.

Reply:
I got it, works much better using your idea.